### DISCUSSION TITLE
github very slow for past 2 weeks

### DISCUSSION THREAD

**Discussion by jeevan-aj** [community/community#73238] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

why is github sooo slow latelty which ever link or button i press it takes minimum 10 seconds to load  .but why is this happening? i have decent internet speed. first i though it was internet speed issue but its not . i tried checking in different times also but still its there why? for posting this i clicked on get started button and it took like  20 seconds to load its just bufferening why?
is it fixable from my side am i doing someting wrong?
if its fixable from my side please suggest solution

**Comment by symroe** [comment]
I've found this too. The interface has become very sluggish since the changes rolled out recently. I assumed this would sort itself out over time, but there seems to be a lot of JS being pushed to the frontend that's slowing things down. 

Can we have a "lite" version, or just a lighter version like the previous one was? Someone said GitHub have switched to React recently, and I think that explains the slowness.

**Reply by hangshan-boop** [reply]
Github: SwitchHosts

**Comment by pluginslab** [comment]
This is on the latest version of Safari only. On Chrome, it works great, was what I could gather.

**Reply by rilysh** [reply]
FYI, this issue also persist on Firefox too.

**Reply by rgandikota** [reply]
Facing this even on Chrome

**Reply by dannyhw** [reply]
Fixed for me by using manual dns on my router. Set to google dns (8.8.8.8) or Cloudflare dns.

**Comment by darkotodoric** [comment]
GitHub has been having issues in recent days. You can see the details on their [status page](https://www.githubstatus.com/history), but in general, it was stated that they had a performance problem (just like you reported).

**Comment by joaomagfreitas** [comment]
+1, also having issues when pulling and pushing with git

**Comment by jmschp** [comment]
My setup

- Google Chrome 120.0.6099.234 (Official Build) (arm64)
- Git 2.42.1
- Apple MacBook Pro 13-inch, M1, 2020
- macOS Sonoma 14.0

With my company profile, working in GitHub Enterprise, sometimes, actually too many times, both navigation on github.com, and pushing and pulling with SSH, is extremely slow.

The slowness is not constant, sometimes a `git push` works just fine, but the next `git push` might take a long time (more than 30 seconds).

The same happens navigating between pages (repos, pull request, etc.). It can take an eternity to load some page, but the next click works just fine. Very inconstant.

But most of the times is just slow, very slow.

I have enabled some of git debugging ENVs
https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables

**Comment by pluginslab** [comment]
I had the same experience. I'm based in Porto, Portugal, and I thought it could be related to some routing of the networks for some reason, but with VPN I have the same "sometimes fast, mostly slow" kind of behavior. Similar setup as well. I mentioned earlier here that it was fine with Chrome, but that was just me hitting the "sometimes fast" side of the problem.

**Comment by joaomagfreitas** [comment]
The still issue persists here in Porto, Portugal
Starting to think this is a network cluster problem on GitHub end or our ISPs are throttling GitHub access

**Reply by dannyhw** [reply]
wow I thought I was going crazy... github really is slow in porto for some reason?

**Reply by joaomagfreitas** [reply]
@dannyhw which ISP are you working with?

**Reply by dannyhw** [reply]
I have vodafone, just tested using a vpn to lisbon and it actually worked.

**Reply by joaomagfreitas** [reply]
I'm also using Vodafone! Will ask friends who use MEO or NOS if they have the same issue

**Reply by jmschp** [reply]
I am also in Porto with Vodafone!

**Reply by Comum** [reply]
+1 for slowness with Vodafone in Porto.
Do you guys also have trouble when pushing to remote? Sometimes I need to re-kick it 2 or 3 times.

**Reply by dannyhw** [reply]
I would have to cancel and restart everything multiple times, now I started using a vpn to lisbon at all times and I don't have to do that.

**Reply by joaomagfreitas** [reply]
> Do you guys also have trouble when pushing to remote? Sometimes I need to re-kick it 2 or 3 times.

exactly what's happening to me
seems like this is a Vodafone issue...

**Reply by dannyhw** [reply]
I've fixed this for my case by adding manual dns on my router to use google dns 8.8.8.8 and 8.8.4.4. Seems like vodafone dns is broken.

**Reply by rilysh** [reply]
@dannyhw have you tried with Cloudflare DNS?

**Reply by dannyhw** [reply]
I believe that would also work, and probably is slightly faster.

**Comment by amonras** [comment]
I started experiencing those same issues since yesterday, after I used Codespaces for the first time

**Comment by SrslyJosh** [comment]
I've had the same problem for months and months, with both public GH and a GH enterprise instance I have access to. Clicking an internal (to GH) link in Safari results in a very noticeable, multi-second delay where the browser doesn't appear to do anything, then the expected page content is rendered.

It goes like this:
1. Click internal link
2. Blue "loading" bar appears at the top of the viewport
3. Loading bar animates normally, the animation completes, and the loading bar disappears
4. The address bar and page title change (at the same instant that the loading bar disappears)
5. Several seconds pass
6. Finally, the new page content is rendered

It's really annoying and makes the whole site feel broken.

**Reply by rilysh** [reply]
This happens on Firefox as well. Sadly GitHub seems as always unaware of these issues.

**Reply by Antti-L** [reply]
I have been seeing this issue occurring regularly irregularly since a long time now, with MS Edge and Firefox.  Very annoying indeed.

**Comment by ambaca** [comment]
Profile page is very, very slow. It is loading page infinite but nothing shows.
